kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -41,6 +43,7 @@ public class SnapshotImpl implements Snapshot {
     private final Protocol protocol;
     private final Metadata metadata;
     private final LogSegment logSegment;
+    private Optional<Long> inCommitTimestampOpt;
```

Comment:
Use `Lazy` instance to load it atmost once. See the examples of `Lazy` in `Metadata.java`